cdap-api/src/main/java/co/cask/cdap/api/schedule/DataSchedule.java

```diff
@@ -0,0 +1,106 @@
+/*
+ * Copyright © 2015 Cask Data, Inc.
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License"); you may not
+ * use this file except in compliance with the License. You may obtain a copy of
+ * the License at
+ *
+ * http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS, WITHOUT
+ * WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied. See the
+ * License for the specific language governing permissions and limitations under
+ * the License.
+ */
+
+package co.cask.cdap.api.schedule;
+
+/**
+ *
+ */
+public class DataSchedule extends Schedule {
+  /**
+   * Source type, in case the {@link ScheduleType} of this schedule is {@code DATA}.
+   */
+  public enum SourceType { STREAM, DATASET }
```

Comment:
What I am worried about is that this in API and we want to change that as little as possible after it is public. 

My gut feeling is that the broader type is a "NotificationSchedule" or something named similarly. Then there can be different types of notification which result in multiple subclasses of NotificationSchedule. Have you thought about what will be the API to specify a mixed time/notification schedule? (schedule whenever there is at least 1GB of data, but at least once per hour).